### PR TITLE
monitoring: make email_delivery_failures on percentage, not count

### PR DESCRIPTION
**Pull request by bobheadxi**
This changes the threshold to critical-alert on 10% failure rate, and warn on any non-zero failure rate (as all email delivery failures impact user experience).

## Test plan

Test query (dotcom): https://sourcegraph.com/-/debug/grafana/explore?orgId=1&left=%5B%22now-2d%22,%22now%22,%22Prometheus%22,%7B%22exemplar%22:true,%22expr%22:%22sum(increase(src_email_send%7Bsuccess%3D%5C%22false%5C%22%7D%5B30m%5D))%20%2F%20sum(increase(src_email_send%5B30m%5D))%20*%20100%22,%22requestId%22:%22Q-c1441862-6745-442a-9745-ba1bf9d80a4e-0A%22%7D%5D&right=%5B%22now-2d%22,%22now%22,%22Prometheus%22,%7B%22exemplar%22:true,%22expr%22:%22sum(increase(src_email_send%7Bsuccess%3D%5C%22false%5C%22%7D%5B30m%5D))%22,%22requestId%22:%22Q-4b1e1c07-aa04-4f9b-ab5d-fef8100a5b24-0Q-c8739f02-df38-476b-8cb0-fc9ddc0fd71c-1A%22%7D,%7B%22exemplar%22:true,%22expr%22:%22sum(increase(src_email_send%5B30m%5D))%22,%22requestId%22:%22Q-4b1e1c07-aa04-4f9b-ab5d-fef8100a5b24-0Q-c8739f02-df38-476b-8cb0-fc9ddc0fd71c-1B%22%7D%5D

Test dashboard: follow https://docs.sourcegraph.com/dev/how-to/monitoring_local_dev#grafana to point to Sourcegraph.com:

image

## Preview 🤩
[Preview Link](https://docs.sourcegraph.com/@monitoring-emails-percentage-failures)

### CODE REVIEW

**Review by unknwon — APPROVED** [review]

**Review by filiphaftek — APPROVED** [review]
Thank you!

File: doc/admin/observability/alerts.md [hunk 1]
```diff
@@ -1216,18 +1216,19 @@ Generated query for critical alert: `min((sum by (app) (up{app=~".*(frontend|sou
...
 ## frontend: email_delivery_failures
 
-<p class="subtitle">email delivery failures every 30 minutes</p>
+<p class="subtitle">email delivery failures over 30 minutes</p>
 
 **Descriptions**
 
-- <span class="badge badge-warning">warning</span> frontend: 1+ email delivery failures every 30 minutes
-- <span class="badge badge-critical">critical</span> frontend: 2+ email delivery failures every 30 minutes
+- <span class="badge badge-warning">warning</span> frontend: 0%+ email delivery failures over 30 minutes
```
Comment (rafax) [inline doc/admin/observability/alerts.md:1223]:
```suggestion
- <span class="badge badge-warning">warning</span> frontend: 0%+ email delivery failure rate over 30 minutes
```

File: doc/admin/observability/alerts.md [hunk 2]
```diff
@@ -1216,18 +1216,19 @@ Generated query for critical alert: `min((sum by (app) (up{app=~".*(frontend|sou
...
+- <span class="badge badge-critical">critical</span> frontend: 10%+ email delivery failures over 30 minutes
```
Comment (rafax) [inline doc/admin/observability/alerts.md:1224]:
```suggestion
- <span class="badge badge-critical">critical</span> frontend: 10%+ email delivery failure rate over 30 minutes
```

File: doc/admin/observability/alerts.md (hunk 1 above)
Comment (bobheadxi) [reply doc/admin/observability/alerts.md:1223]: these are generated from the Go file, so updating from there :)

Comment (bobheadxi) [reply doc/admin/observability/alerts.md]: Fixed in https://github.com/sourcegraph/sourcegraph/pull/57045/commits/58a186f95186b8f4eccbcd1917205d211ff5ed7a!

**Pull request merged**